### ISSUE TITLE
How to toggle off stripe correction on the command line?

### ISSUE THREAD

**Issue by gmorenomello** [nclack/whisk#45]
I need to process many files and I am having a problem that whiki is creating stripe artifacts instead of removing it. Now, when I go in whiski I can eliminate it by clicking "s". How can I do the same with the command line?

**Comment by nclack** [comment]
Unfortunately, it looks like this requires changing some code:

This [line](https://github.com/nclack/whisk/blob/d938b9d8dc332d0494cbced6ae844c0ff8250b46/src/whisk.c#L59):

      TRY(im=video_get(v,index,1),ErrorRead);

needs to be changed to this:

     TRY(im=video_get(v,index,0),ErrorRead);

And then you have to rebuild everything.  It can be hard to build the code.  What operating system are you using?

**Comment by nclack** [comment]
In case it helps, the README has some instructions on building.

**Comment by gmorenomello** [comment]
Windows 10 and MacOsX

**Comment by gmorenomello** [comment]
Ok I will try
Thanks

> Em 25 de set de 2018, à(s) 14:56, Nathan Clack <notifications@github.com> escreveu:
> 
> the README ha

**Comment by ninjasatwork** [comment]
I tried to compile it in the MACOSX and received the following error message:
-- Found unsuitable Qt version "5.6.2" from /Users/gbmello/anaconda3/bin/qmake
searching /Applications/MATLAB_R2016b.app
Searching /Applications/MATLAB_R2016b.app ... /Applications/MATLAB_R2016b.app/bin/glnxa64
Searching /Applications/MATLAB_R2016b.app ... /Applications/MATLAB_R2016b.app/bin/maci64
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
OGG_LIBRARY (ADVANCED)
    linked by target "vsplit" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "totif" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "seedtest" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "preprocess" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "trace" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "stripetest" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "whiskio" in directory /Users/gbmello/Dropbox/CODE/whisk-master
    linked by target "whisk" in directory /Users/gbmello/Dropbox/CODE/whisk-master

-- Configuring incomplete, errors occurred!
See also "/Users/gbmello/Dropbox/CODE/whisk-master/buildMac/CMakeFiles/CMakeOutput.log".
(base) 1x-193-157-207-112:buildMac gbmello$ 

any ideas on how should I set them? and what QT is required?

**Comment by gmorenomello** [comment]
Hey Nathan,

I tried to follow your recommendations. I tried to compile first in MacOsX Sierra and then in Windows 10. Both of them failed.

The code seems to be looking for variables that I am supposed to have. In mac you can see the previous post. In PC (Windows 10) the message is as follows:

> -- Building for: Visual Studio 15 2017
> -- Selecting Windows SDK version 10.0.17134.0 to target Windows 10.0.16299.
> -- The C compiler identification is MSVC 19.14.26433.0
> -- The CXX compiler identification is MSVC 19.14.26433.0
> -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe
> -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe
> -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> awk not found.  Using previously generated mylib c files in C:/Users/gbmello/Dropbox/environments/whisk/generated.
> MYLIB_SRCS is C:/Users/gbmello/Dropbox/environments/whisk/src/generated/contour_lib.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/draw_lib.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/image_filters.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/image_lib.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/level_set.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/utilities.c;C:/Users/gbmello/Dropbox/environments/whisk/src/generated/water_shed.c
> -- Could NOT find BISON (missing: BISON_EXECUTABLE)
> Bison not found.  Using previously generated parameter parser.
> -- Found unsuitable Qt version "" from NOTFOUND
> searching C:/Program Files/Matlab/R2017a
> Searching C:/Program Files/Matlab/R2017a ... C:/Program Files/Matlab/R2017a/extern/lib/win32/microsoft/msvc71
> Searching C:/Program Files/Matlab/R2017a ... C:/Program Files/Matlab/R2017a/extern/lib/win32/microsoft
> -- Configuring done
> -- Generating done
> -- Build files have been written to: 
> 
>  C:/Users/gbmello/Dropbox/environments/whisk/build

After this, when I try to run `make` it says that there is no such file.
It doesnt find Bison, I added it to the system environment and nothing.
Also it is complaining about the version of QT and I have no idea which one to use. I have the Qt5.2 installed
 I have installed Bison, Gawk, Cmake and QT. I ran out of ideas.

Could you please point me out what am I doing wrong?

**Comment by nclack** [comment]
On windows, cmake will  generate a visual studio solution (a .sln file)  instead of a makefile.  See [here](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html?highlight=generators) for more info.  You'll need [visual studio](https://visualstudio.microsoft.com/downloads/) installed to build the software.

The good news is it looks like the cmake step worked.  If you see these lines at the end, that's a good sign:

```
  -- Configuring done
  -- Generating done
```

You can run [cmake-gui](https://cmake.org/cmake/help/latest/manual/cmake-gui.1.html) to inspect settings that cmake used during configuration.  For example if you search for BISON, you'll find the variable used and you can point that to where you installed it  (It doesn't look like you _need_ bison though).

**Comment by gmorenomello** [comment]
Hey Nathan, I still didn’t managed to run it in visual studio. I feel like I might be missing something really dumb do you have any suggestion?

**Comment by gmorenomello** [comment]
So I managed to make it work. Nonetheless the tracking that it provides me is still consistent with the tracking with the stripe correction. I tried to open it in Whiski to check. And when I remove the stripe correction there and ask it to trace, the whiskers are traced correctly. So it is unlikely to be the image itself.

Is there anywhere else that I might be missing to change in the code?

**Comment by nclack** [comment]
Is there way you can show me what you changed so I can double check? 

I'll link you to the functions where the strip correction is done so you can verify  (via breakpoints or printfs etc) whether they are still being called.  They are [here](https://github.com/nclack/whisk/blob/d938b9d8dc332d0494cbced6ae844c0ff8250b46/src/adjust_scan_bias.c#L152) and [here](https://github.com/nclack/whisk/blob/d938b9d8dc332d0494cbced6ae844c0ff8250b46/src/adjust_scan_bias.c#L177).

**Comment by gmorenomello** [comment]
Is there a way I can send you a small mp4 clip so you can check if you can trace it?

**Comment by gmorenomello** [comment]
in whisk.c

**Comment by gmorenomello** [comment]
the parts of the code responsible for the vertical lines are not being executed. can you just double check With Your machine? you might get a better Insight of why I cannot properly trace this https://www.dropbox.com/s/4qgy41jygwkx1vw/20180730_1210_12-10-21.000_part_1.mp4?dl=0

**Comment by nclack** [comment]
There's something strange about the timing information in the mpg.    I think whisk isn't seeking around to the correct frames.  If I seek forward in whiski, the video looks fine, but it doesn't seek backwards correctly. This [thread](https://github.com/nclack/whisk/issues/35#issuecomment-276996644) might help.  You might consider extracting a small part of the video to a tiff stack and checking if that changes anything.

That said, I can get the tracing to find some of the trimmed whiskers near the face, but the untrimmed whisker gives me trouble.  There's a set of parameters saved in the `default.parameters` file.  That file either lives in the current working directory or in your install directory (`C:\Program Files\WhiskerTracking`).  I was able to get some success by changing the `HALF_SPACE_ASSYMETRY_THRESH` to 1.0.  You have a slight background gradient from left to right across the image.  I thought that might be tripping this threshold.  It's used to stop tracing near the face or near collisions with other objects.

I'm not quite sure, but I think the noisy line on the right side of the image might was causing the "stripe adjustment" to be especially bad.  Turning the stripe-adjust off is the correct thing to do as your camera produces beautiful images (save for that one line).

**Comment by gmorenomello** [comment]
This video seems to track ok. But here is what I found in other video.
![screenshot 5](https://user-images.githubusercontent.com/43554437/47098318-47871400-d233-11e8-81a4-6b9d08c78b4d.png)

Any idea? Also, it seems that it tracks the whisker with less mistakes if the tip of the whisker is visible. Is that part of the algorithm somewhere? I missed in the paper.

**Comment by tvajtay** [comment]
I was able to track fine without seeing the whisker tip, I don't believe it is needed for segmentation.
One of my kludge fixes for video artifacts was to do a Gaussian blur and increase contrast in imagej so I could retain the general whisker profile while smudging out the artifacts. I don't know if that's relevant here.

**Comment by gmorenomello** [comment]
It puzzles me, why would I have Horizontal lines now?  @nclack  do you have any hypothesis?
also I changed the default.parameter that is inside the build folder (I use the c functions directly from there) and observed no difference in tracing.
horizontal error file.
https://www.dropbox.com/sh/zz2e2dzrrr650g3/AACySO4tHJhNg6RkgG6Tr5Sua?dl=0

**Comment by gmorenomello** [comment]
I tryied tvajtay's solution and worked like a charm. I had to use the gaussian blur and the adaptive histogram equalization. With these two I was able to improve tracking. Here is the caveat. MP4 files actually (for whatever reason) make trigger something in the algorithm that makes it to track parallel lines, either horizontal or vertical (I still do not know why one over the other). For his solution to work it is necessary to extract the videos from SEQ to uncompressed avi.  In this case it seems to be working. I will run some more files and confirm whether this solution is stable.
